specs/ProcessInfo.md

```diff
@@ -0,0 +1,269 @@
+            CHECK_FAILURE(processInfo->get_Kind(&kind));
+
+            WCHAR id[4096] = L"";
+            StringCchPrintf(id, ARRAYSIZE(id), L"Process ID: %u", processId);
+
+            HANDLE processHandle = OpenProcess(PROCESS_ALL_ACCESS, FALSE, processId);
+            PROCESS_MEMORY_COUNTERS_EX pmc;
+            GetProcessMemoryInfo(
+                processHandle, reinterpret_cast<PROCESS_MEMORY_COUNTERS*>(&pmc), sizeof(pmc));
+            SIZE_T virtualMemUsed = pmc.PrivateUsage / 1024;
+            WCHAR memory[4096] = L"";
+            StringCchPrintf(memory, ARRAYSIZE(memory), L"Memory: %u", virtualMemUsed);
+
+            result = result + id + L" | Process Kind: " + ProcessKindToString(kind) + L" | " +
+                     memory + L" KB\n";
+        }
+    }
+    MessageBox(nullptr, result.c_str(), L"Memory Usage", MB_OK);
+}
+//! [ProcessInfoChanged]
+```
+
+# API Details    
+```
+interface ICoreWebView2Environment6;
+interface ICoreWebView2ProcessInfo;
+interface ICoreWebView2ProcessInfoCollection;
+interface ICoreWebView2ProcessInfoChangedEventHandler;
+
+[v1_enum]
+typedef enum COREWEBVIEW2_PROCESS_KIND {
+  /// Indicates the browser process kind.
+  COREWEBVIEW2_PROCESS_KIND_BROWSER,
+
+  /// Indicates the render process kind.
+  COREWEBVIEW2_PROCESS_KIND_RENDERER,
+
+  /// Indicates the utility process kind.
+  COREWEBVIEW2_PROCESS_KIND_UTILITY,
+
+  /// Indicates the sandbox helper process kind.
+  COREWEBVIEW2_PROCESS_KIND_SANDBOX_HELPER,
+
+  /// Indicates the GPU process kind.
+  COREWEBVIEW2_PROCESS_KIND_GPU,
+
+  /// Indicates the PPAPI plugin process kind.
+  COREWEBVIEW2_PROCESS_KIND_PPAPI_PLUGIN,
+
+  /// Indicates the PPAPI plugin broker process kind.
+  COREWEBVIEW2_PROCESS_KIND_PPAPI_BROKER,
+
+  /// Indicates the process of unspecified kind.
+  COREWEBVIEW2_PROCESS_KIND_UNKNOWN,
+} COREWEBVIEW2_PROCESS_KIND;
+
+[uuid(20856F87-256B-41BE-BD64-AB1C36E3D944), object, pointer_default(unique)]
+interface ICoreWebView2Environment6 : ICoreWebView2Environment5
+{
+  /// Adds an event handler for the `ProcessInfoChanged` event.
+  /// 
+  /// \snippet ProcessComponent.cpp ProcessInfoChanged
+  HRESULT add_ProcessInfoChanged(
+      [in] ICoreWebView2ProcessInfoChangedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_ProcessInfoChanged`.
+  HRESULT remove_ProcessInfoChanged(
+      [in] EventRegistrationToken token);
+
+  /// Returns the `ICoreWebView2ProcessInfoCollection`
+  /// Provide a list of all process using same user data folder.
+  [propget] HRESULT ProcessInfo([out, retval]ICoreWebView2ProcessInfoCollection** value);
+}
+
+/// Provides a set of properties for a process in the `ICoreWebView2Environment`.
+[uuid(7798D399-52A1-4823-AD6A-1F3EDD74B0B6), object, pointer_default(unique)]
+interface ICoreWebView2ProcessInfo : IUnknown {
+
+  /// The process id of the process.
+  [propget] HRESULT Id([out, retval] UINT32* value);
```

Comment:
I think `ProcessId` is a clearer name than just `Id`. A plain `Id` is usually a string identifier.

`Windows.System.Diagnostics.ProcessDiagnosticInfo` calls it `ProcessId`, not `Id`.